frontend/packages/local-storage-operator-plugin/src/components/disks-list/disks-list-page.tsx

```diff
@@ -47,66 +48,89 @@ export const tableColumnClasses = [
 ];
 
-const DiskHeader = () => {
+export const DiskHeader = (disableItems = {}) => () => {
   return [
     {
       title: 'Name',
       sortField: 'path',
       transforms: [sortable],
       props: { className: tableColumnClasses[0] },
     },
+    {
+      title: 'Node',
+      sortField: 'node',
+      transforms: [sortable],
+      props: { className: tableColumnClasses[1] },
+    },
     {
       title: 'Disk State',
       sortField: 'status.state',
       transforms: [sortable],
-      props: { className: tableColumnClasses[1] },
+      props: { className: tableColumnClasses[2] },
     },
     {
       title: 'Type',
       sortField: 'type',
       transforms: [sortable],
-      props: { className: tableColumnClasses[2] },
+      props: { className: tableColumnClasses[3] },
     },
     {
       title: 'Model',
       sortField: 'model',
       transforms: [sortable],
-      props: { className: tableColumnClasses[3] },
+      props: { className: tableColumnClasses[4] },
     },
     {
       title: 'Capacity',
       sortField: 'size',
       transforms: [sortable],
-      props: { className: tableColumnClasses[4] },
+      props: { className: tableColumnClasses[5] },
     },
     {
       title: 'Filesystem',
       sortField: 'fstype',
       transforms: [sortable],
-      props: { className: tableColumnClasses[5] },
+      props: { className: tableColumnClasses[6] },
     },
-  ];
+  ].filter((item) => {
+    return !disableItems[item.title];
+  });
 };
 
-const DiskRow: RowFunction<DiskMetadata> = ({ obj, index, key, style }) => {
+export const DiskRow: RowFunction<DiskMetadata> = ({ obj, index, key, style, customData }) => {
   return (
     <TableRow id={obj.deviceID} index={index} trKey={key} style={style}>
       <TableData className={tableColumnClasses[0]}>{obj.path}</TableData>
-      <TableData className={tableColumnClasses[1]}>{obj.status.state}</TableData>
-      <TableData className={tableColumnClasses[2]}>{obj.type || '-'}</TableData>
-      <TableData className={cx(tableColumnClasses[3], 'co-break-word')}>
+      {!customData?.disableItems?.Node && (
+        <TableData className={tableColumnClasses[1]}>{obj.node}</TableData>
+      )}
+      {!customData?.disableItems?.['Disk State'] && (
+        <TableData className={tableColumnClasses[2]}>{obj.status.state}</TableData>
+      )}
+      <TableData className={tableColumnClasses[3]}>{obj.type || '-'}</TableData>
+      <TableData className={cx(tableColumnClasses[4], 'co-break-word')}>
         {obj.model || '-'}
       </TableData>
-      <TableData className={tableColumnClasses[4]}>
+      <TableData className={tableColumnClasses[5]}>
         {humanizeBinaryBytes(obj.size).string || '-'}
       </TableData>
-      <TableData className={tableColumnClasses[5]}>{obj.fstype || '-'}</TableData>
+      {!customData?.disableItems?.Filesystem && (
+        <TableData className={tableColumnClasses[6]}>{obj.fstype || '-'}</TableData>
+      )}
```

Comment:
Its just two functions `DiskRow` and `DiskHeader` .
Can you consider re-creating them for the time being since most of the items instead of being reused are on `checks` to show or not.
I am working on making it extensible, we can improve in future.